website/public/js/controllers/groupsCtrl.js

```diff
@@ -308,6 +308,11 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       return message.highlight;
     }
 
+    $scope.togglehidingSystemMessages = function() {
+      // Intentionally flip the boolean state
+      $scope.hideSystemMessages = $scope.hideSystemMessages ? false : true;
```

Comment:
This could be simplified as: `$scope.hideSystemMessages = !$scope.hideSystemMessages`

I find that less confusing to look at then a ternary operator and I don't think the comment will be necessary.